### ISSUE TITLE
‘recurrent’ in the paper doesn't seem to involve LSTM

### ISSUE THREAD

**Issue by futureer** [airalcorn2/Recurrent-Convolutional-Neural-Network-Text-Classifier#8]
Hi, thanks for sharing your implementation of the paper "RCNN for text classification". I cloned this repo and experimented on my text classification task, the performance didn't behave as expected. I am not sure it happened due to my data pre-processing or the model implementation.

And after looking through the paper, I found that  Equation (1), (2) give the computation of cl and cr:
![image](https://user-images.githubusercontent.com/3355365/28326796-531dffda-6c14-11e7-80e6-06d6ef3bdc22.png)

where `cr` and `cl` are the result of simply matrix multiplication and an activation function. But in this code, matrix multiplications are replaced with LSTM cells. Is that proved effective than the original one?

thanks.

**Comment by airalcorn2** [comment]
Yes, I [explicitly state](https://github.com/futureer/Recurrent-Convolutional-Neural-Network-Text-Classifier/blob/master/recurrent_convolutional_keras.py#L37) I'm diverging from the paper in using LSTMs instead of vanilla RNNs, but it's trivial to switch to vanilla RNNs if you wish:

```python
forward = SimpleRNN(hidden_dim_1, return_sequences = True)(l_embedding)
backward = SimpleRNN(hidden_dim_1, return_sequences = True, go_backwards = True)(r_embedding)
```

However, LSTMs will nearly always outperform vanilla RNNs because of the [exploding/vanishing gradient problem](http://proceedings.mlr.press/v28/pascanu13.pdf). I can also confirm that the model works on data sets I've encountered at Red Hat.
